fastapi/dependencies/utils.py

```diff
@@ -245,15 +245,31 @@ def is_scalar_sequence_field(field: ModelField) -> bool:
 def get_typed_signature(call: Callable[..., Any]) -> inspect.Signature:
     signature = inspect.signature(call)
     globalns = getattr(call, "__globals__", {})
-    typed_params = [
-        inspect.Parameter(
-            name=param.name,
-            kind=param.kind,
-            default=param.default,
-            annotation=get_typed_annotation(param, globalns),
-        )
-        for param in signature.parameters.values()
-    ]
+    if not inspect.isclass(call):
+        typed_params = [
+            inspect.Parameter(
+                name=param.name,
+                kind=param.kind,
+                default=param.default,
+                annotation=get_typed_annotation(param, globalns),
+            )
+            for param in signature.parameters.values()
+        ]
+    else:
+        from fastapi import Query
+        fields = {}
+        if call.__dict__.get("__fields__"):
+            for param in call.__fields__:
+                fields[param] = dict((call.__fields__[param].field_info.__repr_args__()))
+        typed_params = [
+            inspect.Parameter(
+                name=param.name,
+                kind=param.kind,
+                default=Query(fields[param.name].get("default"), description=fields[param.name].get("description")),
+                annotation=get_typed_annotation(param, globalns),
+            )
+            for param in signature.parameters.values()
+        ]
```

Comment:
Also, you check the reversed condition and else. It makes it hard to read. Another thing is typed params should be defined about of if-else statement because only one parameter is different, the different parameter should be defined in the conditions. I have amended it as below.
```py
if inspect.isclass(call):
    from fastapi import Query
    fields = {}
    for param in getattr(call, '__fields__', []):
        fields[param] = dict((call.__fields__[param].field_info.__repr_args__()))

    default = Query(fields[param.name].get("default"), description=fields[param.name].get("description"))
else:
    default = param.default

typed_params = [
    inspect.Parameter(
        name=param.name,
        kind=param.kind,
        default=Query(fields[param.name].get("default"), description=fields[param.name].get("description")),
        annotation=get_typed_annotation(param, globalns),
    )
    for param in signature.parameters.values()
]

```

Reply:
I've checked this code.
It makes no sense. `param` is inner variable of `for statement` but you're using this out of scope. This code is totally disordered.

And, if variable `call` is a class, `__fields__` returns dict not list.

I think you never test this code.
And most of cases, `call` is not a class so, giving condition at `default` is a bad way.

Reply:
Additionally, this code doesn't work.

Reply:
> And, if variable call is a class, __fields__ returns dict not list.

OK. Then it should be

```py
for param in getattr(call, '__fields__', {}):
```



> param is inner variable of for statement but you're using this out of scope. This code is totally disordered.
> And most of cases, call is not a class so, giving condition at default is a bad way.

I do not get it.

Reply:
> param is inner variable of for statement but you're using this out of scope. This code is totally disordered.

Okay, I got it. 

Please do not move `typed_params` out of conditions.

Reply:
If `parameters` dict contain `param.name` value all the time, then go ahead with the following 

```py
default=Query(
    fields.get(param.name, {}).get("default"),
    description=fields.get(param.name, {}).get("description")
)
```

Reply:
> Does `parameters` dict contain `param.name` value all the time?

Yes it always contains `param.name`.

Do you know what things are happening when endpoint depends schema? Would you please reply after test yourself and how it works exactly?

Reply:
Okay, I tried to test but could not reproduce how `call ` can be a class. 

By the way, [Sourcery](https://sourcery.ai/) AI suggests the below implementation about `parameters` dict.

```py
parameters = {
    param: dict((fields[param].field_info.__repr_args__()))
    for param in fields
}
```